scripts/policy/frameworks/cluster/agent/main.zeek

```diff
@@ -9,18 +9,18 @@
 redef ClusterController::role = ClusterController::Types::AGENT;
 
 # The global configuration as passed to us by the controller
-global global_config: ClusterController::Types::Configuration;
+global g_config: ClusterController::Types::Configuration;
```

Comment:
Yeah that's nice. Atm there are globals for the agent's runtime (`main.zeek`), the controller's, and for the request abstraction in the `ClusterController::Request` module. I'll keep an eye on it.

Btw, I at times wondered whether it'd be good to move code that both agent and controller rely on to a separate location. I was torn about naming so didn't do it. For example, requests live in `ClusterController::Requests`, but it's also used by the agent to track queries out to the supervisor and back. It might be worth pondering a namespace layout that accommodates the management cluster (what we're building here) and the data cluster ("the cluster" so far, `Cluster::`), to make this more natural. Just to throw something out there: we could use `Management::Controller` and `Management::Agent`, and then have e.g. `Management::Requests` not tied to controller or agent. I also like "management framework" as a name for this whole thing.

Reply:
> Btw, I at times wondered whether it'd be good to move code that both agent and controller rely on to a separate location. 

Good point. I had been wondering about that too, but couldn't quite pinpoint yet what to suggest; same for the naming on existing cluster framework vs the new controller layer.

> Just to throw something out there: we could use `Management::Controller` and `Management::Agent`, and then have e.g. `Management::Requests` not tied to controller or agent. I also like "management framework" as a name for this whole thing.

I like "management". Also, I've been starting to wonder if "ClusterController" is too confusing given the existing ZeekControl. So, yes, could work well to refer to the whole thing as "mgmt framework" and the controller being just an internal piece of it.